app/controllers/events_controller.rb

```diff
@@ -0,0 +1,52 @@
+class EventsController < ApplicationController
+  before_action :logged_in_user, only: [:checkin, :create]
+  before_action :find_event, only: [:checkin, :assistents]
+
+  def index
+    @eventspast = Event.past
+    @eventsupcoming = Event.upcoming
+  end
+
+  def new
+    @event = Event.new
+  end
+
+  def create
+    @event = current_user.events_as_owner.build(event_params)
+
+    if @event.save
+      @event.create_owner_relationship(user_id: current_user.id, role: 'owner')
+
+      flash[:info] = 'Successfully created an event'
+      redirect_to :root
+    else
+      render 'new'
+    end
+  end
+
+  def checkin
```

Comment:
`checkin` and `assistents` methods as said are part of Attendance model (both of them are making records in the Attendance table with event.attneddace build option). Please move both of the methods to the appropriate controller.

Reply:
moved those methods.